src/abstract/ZoweTreeProvider.ts

```diff
@@ -82,52 +82,12 @@ export class ZoweTreeProvider {
      */
     public async flipState(element: IZoweTreeNode, isOpen: boolean = false) {
         element.collapsibleState = isOpen ? vscode.TreeItemCollapsibleState.Expanded : vscode.TreeItemCollapsibleState.Collapsed;
```

Comment:
The equivalent code which moved to `checkCurrentProfile` is no longer being run here. Is it not needed here because it's now being handled in other locations where that function is called?

Reply:
Yes, this fixed the bugs Jelly mentioned. Apparently whenever a node is clicked, vscode finds the tree provider and calls its `getChildren` function. So with the profile check on this function, the password is prompted from `flipState` but the tree is still expanded from `getChildren`.

To fix it I moved the `checkCurrentProfile` call to `getChildren` and removed it from `flipState` so it wouldn't be called twice.